edalize/symbiflow.py

```diff
@@ -52,31 +59,138 @@ def get_doc(cls, api_ver):
                     {
                         "name": "pnr",
                         "type": "String",
-                        "desc": 'Place and Route tool. Currently only "vpr" is supported',
+                        "desc": 'Place and Route tool. Currently only "vpr"/"vtr" and "nextpnr" are supported',
                     },
                     {
-                        "name": "vpr_options",
+                        "name": "pnr_options",
                         "type": "String",
-                        "desc": "Additional vpr tool options. If not used, default options for the tool will be used",
+                        "desc": "Additional options for PnR tool. If not used, default options for the tool will be used",
                     },
-                    {
-                        "name" : "environment_script",
-                        "type" : "String",
-                        "desc" : "Optional bash script that will be sourced before each build step."
-                    },
-                ]
+                ],
             }
 
             symbiflow_members = symbiflow_help["members"]
 
             return {
-                "description": "The Symbiflow backend executes Yosys sythesis tool and VPR place and route. It can target multiple different FPGA vendors",
+                "description": "The Symbiflow backend executes Yosys sythesis tool and VPR/Nextpnr place and route. It can target multiple different FPGA vendors",
                 "members": symbiflow_members,
             }
 
     def get_version(self):
         return "1.0"
 
+    def configure_nextpnr(self):
+        (src_files, incdirs) = self._get_fileset_files(force_slash=True)
+        vendor = self.tool_options.get("vendor")
+
+        # Yosys configuration
+        yosys_synth_options = self.tool_options.get("yosys_synth_options", "")
+        yosys_template = self.tool_options.get("yosys_template")
+        yosys_edam = {
+                "files"         : self.files,
+                "name"          : self.name,
+                "toplevel"      : self.toplevel,
+                "parameters"    : self.parameters,
+                "tool_options"  : {
+                                    "yosys" : {
+                                        "arch" : vendor,
+                                        "yosys_synth_options" : yosys_synth_options,
+                                        "yosys_template" : yosys_template,
+                                        "yosys_as_subtool" : True,
+                                    }
+                                }
+                }
+
+        yosys = getattr(import_module("edalize.yosys"), "Yosys")(yosys_edam, self.work_root)
+        yosys.configure()
+
+        # Nextpnr configuration
+        arch = self.tool_options.get("arch")
+        if arch not in getattr(self, "archs"):
```

Comment:
Nitpick: Use `if arch not in self.archs:`

Reply:
fixed